### PR TITLE
Feature/browserstack

### LINKED ISSUE #1355: Run UI tests with Browserstack
Testcafe can connect to Browserstack. Its not active because there are problems wie IE. So fix that to use that nice feature.

### PR DESCRIPTION
**Pull request by danyball**
Fixes #1355 

# Done is when (DoD):
- [ ] Modifications within components are reflected by tests.
- [ ] A self-review of the code has been performed.
- [ ] The modified component properties have been added to the typescript typings.
- [ ] Potential design changes have been reviewed by a designer.
- [ ] The modifications are well documented (README.md, etc.) and showcased in the stories.
- [ ] The modifications are documented in the code, particularly in hard-to-understand areas (focus on **WHY**).
- [ ] The modifications are shortly added to CHANGELOG.md with the issue number.
- [ ] The modifications still work in IE.
- [ ] The modifications are according to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,17 +15,18 @@
...
     "test-jest-watch": "jest --env=jsdom --watch -o",
     "test-ui": "TEST_HOST_STORYBOOK_URL=http://localhost:6006 testcafe chrome ./**/ui.test.js",
-    "test-ui-hl": "TEST_HOST_STORYBOOK_URL=http://localhost:6006 testcafe 'chrome:headless' ./**/ui.test.js",
+    "test-ui-headless": "TEST_HOST_STORYBOOK_URL=http://localhost:6006 testcafe 'chrome:headless' ./**/ui.test.js",
     "new": "node scripts/create-a-m-o/index.js",
-    "deploy-branch": "npm run build-storybook && SOURCE_BRANCH_NAME=\"$(git rev-parse --abbrev-ref HEAD)\" bash scripts/branch-deployment.sh && SOURCE_BRANCH_NAME=\"$(git rev-parse --abbrev-ref HEAD)\" bash scripts/branch-deployment-notifier.sh"
+    "deploy-branch": "npm run build-storybook && SOURCE_BRANCH_NAME=\"$(git rev-parse --abbrev-ref HEAD)\" bash scripts/branch-deployment.sh && SOURCE_BRANCH_NAME=\"$(git rev-parse --abbrev-ref HEAD)\" bash scripts/branch-deployment-notifier.sh",
+    "test:e2e:browserstack": "TEST_HOST_STORYBOOK_URL=http://localhost:6006 BROWSERSTACK_USERNAME=danielb17 BROWSERSTACK_ACCESS_KEY=z9ereHMoK6twktEx84wa testcafe 'browserstack:chrome:Windows' ./**/datepicker/ui.test.js"
```
Comment (raphaellueckl) [inline package.json]: Never publish a key to a public repo (even non-public ones). You have to make this one invalid. There are scripts that automatically scan github, bitbucket, etc... for stuff like that.

**Comment by danyball** [comment]
This is the log of browserstack run of datepicker tests @ `chrome@64.0:OS X High Sierra`:

```
 Running tests in:
 - Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

 Datepicker
 ✓ should select February the 13th and then the 14th
 ✓ should convert the mixed input values (numbers and ranges) from allowedyears prop correctly
 ✓ should select the first of march from within the February view
 ✓ should select the 31th of January from within the February view
 ✓ should have a 29th of February in 2020 - should correctly handle leap year
 ✓ should begin with September 1 as first Monday in 1980
 ✓ should handle month change with native dropdown element

 Datepicker - With Locale
 ✖ should display month in english

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         11 |    const dropDown = await Selector(this.selectMonthDropdownSelector(this.id));
         12 |    await this.t.click(dropDown);
         13 |    const selectedMonth = await Selector(
         14 |      this.selectAnyMonthSelector(this.id, numericMonth, native)
         15 |    );
       > 16 |    await this.t.click(selectedMonth);
         17 |  }
         18 |
         19 |  async chooseAnyYear(numericYear, native) {
         20 |    await this.t.setTestSpeed(0.5);
         21 |    const dropDown = await Selector(this.yearDropdownSelector(this.id));

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/test.accessor.js:16:18)


 Datepicker - Collapsible Version
 ✖ should write date into input field for input calendars

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         11 |    const dropDown = await Selector(this.selectMonthDropdownSelector(this.id));
         12 |    await this.t.click(dropDown);
         13 |    const selectedMonth = await Selector(
         14 |      this.selectAnyMonthSelector(this.id, numericMonth, native)
         15 |    );
       > 16 |    await this.t.click(selectedMonth);
         17 |  }
         18 |
         19 |  async chooseAnyYear(numericYear, native) {
         20 |    await this.t.setTestSpeed(0.5);
         21 |    const dropDown = await Selector(this.yearDropdownSelector(this.id));

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/test.accessor.js:16:18)


 ✖ should change enhanced dropdown title (only on large screens) on month change

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         11 |    const dropDown = await Selector(this.selectMonthDropdownSelector(this.id));
         12 |    await this.t.click(dropDown);
         13 |    const selectedMonth = await Selector(
         14 |      this.selectAnyMonthSelector(this.id, numericMonth, native)
         15 |    );
       > 16 |    await this.t.click(selectedMonth);
         17 |  }
         18 |
         19 |  async chooseAnyYear(numericYear, native) {
         20 |    await this.t.setTestSpeed(0.5);
         21 |    const dropDown = await Selector(this.yearDropdownSelector(this.id));

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/test.accessor.js:16:18)


 Datepicker React
 ✓ should render datepicker as reactified component
 ✓ should correctly expand year ranges assigned via property

 Datepicker React controlled
 ✓ datepicker should behave correctly when controlled

 Datepicker React inputfield
 ✓ should fire onDateChange callback on valid user input

 Datepicker React inputfield
 ✓ should react to programmatic date property changes

 Datepicker React empty inputfield
 ✖ should allow month change from default date

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         399 |  const newMonth = currentMonth + (currentMonth === JANUARY ? 1 : -1);
         400 |
         401 |  // click on new month
         402 |  await t
         403 |    .wait(50)
       > 404 |    .click(
         405 |      `#datepicker-react .js-dropdown__button[data-index="${newMonth + 1}"]`
         406 |    );
         407 |
         408 |  // commit new date
         409 |  await t.wait(50).click(`#datepicker-react .js-datepicker__button-ok`);

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/ui.test.js:404:6)


 Datepicker React empty inputfield
 ✓ should pass through className and arbitrary data- attributes + have automatic React displayName
 ✓ should display error message if invalid is set, unless invaliddatetext is empty

 Datepicker Form
 ✓ should submit datepicker correctly in form
 ✓ should have default width

 Datepicker with onchange handler
 ✓ should fire the right events

 Datepicker as inputfield with fixed width and height
 ✓ should have 200px width
 ✓ button should have correct height
 ✓ input wrap should have correct height without a label

 Datepicker as inputfield with label
 ✓ component and input wrap should have correct height

 Datepicker as inputfield with 196px width and 10px height
 ✓ should have a minimum width
 ✓ should have a minimum height

 Datepicker
 ✖ should select February the 13th and then the 14th

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         11 |    const dropDown = await Selector(this.selectMonthDropdownSelector(this.id));
         12 |    await this.t.click(dropDown);
         13 |    const selectedMonth = await Selector(
         14 |      this.selectAnyMonthSelector(this.id, numericMonth, native)
         15 |    );
       > 16 |    await this.t.click(selectedMonth);
         17 |  }
         18 |
         19 |  async chooseAnyYear(numericYear, native) {
         20 |    await this.t.setTestSpeed(0.5);
         21 |    const dropDown = await Selector(this.yearDropdownSelector(this.id));

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/test.accessor.js:16:18)


 ✓ should convert the mixed input values (numbers and ranges) from allowedyears prop correctly
 ✖ should select the first of march from within the February view

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         11 |    const dropDown = await Selector(this.selectMonthDropdownSelector(this.id));
         12 |    await this.t.click(dropDown);
         13 |    const selectedMonth = await Selector(
         14 |      this.selectAnyMonthSelector(this.id, numericMonth, native)
         15 |    );
       > 16 |    await this.t.click(selectedMonth);
         17 |  }
         18 |
         19 |  async chooseAnyYear(numericYear, native) {
         20 |    await this.t.setTestSpeed(0.5);
         21 |    const dropDown = await Selector(this.yearDropdownSelector(this.id));

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/test.accessor.js:16:18)


 ✖ should select the 31th of January from within the February view

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         11 |    const dropDown = await Selector(this.selectMonthDropdownSelector(this.id));
         12 |    await this.t.click(dropDown);
         13 |    const selectedMonth = await Selector(
         14 |      this.selectAnyMonthSelector(this.id, numericMonth, native)
         15 |    );
       > 16 |    await this.t.click(selectedMonth);
         17 |  }
         18 |
         19 |  async chooseAnyYear(numericYear, native) {
         20 |    await this.t.setTestSpeed(0.5);
         21 |    const dropDown = await Selector(this.yearDropdownSelector(this.id));

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/test.accessor.js:16:18)


 ✖ should have a 29th of February in 2020 - should correctly handle leap year

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         11 |    const dropDown = await Selector(this.selectMonthDropdownSelector(this.id));
         12 |    await this.t.click(dropDown);
         13 |    const selectedMonth = await Selector(
         14 |      this.selectAnyMonthSelector(this.id, numericMonth, native)
         15 |    );
       > 16 |    await this.t.click(selectedMonth);
         17 |  }
         18 |
         19 |  async chooseAnyYear(numericYear, native) {
         20 |    await this.t.setTestSpeed(0.5);
         21 |    const dropDown = await Selector(this.yearDropdownSelector(this.id));

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/test.accessor.js:16:18)


 ✖ should begin with September 1 as first Monday in 1980

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         11 |    const dropDown = await Selector(this.selectMonthDropdownSelector(this.id));
         12 |    await this.t.click(dropDown);
         13 |    const selectedMonth = await Selector(
         14 |      this.selectAnyMonthSelector(this.id, numericMonth, native)
         15 |    );
       > 16 |    await this.t.click(selectedMonth);
         17 |  }
         18 |
         19 |  async chooseAnyYear(numericYear, native) {
         20 |    await this.t.setTestSpeed(0.5);
         21 |    const dropDown = await Selector(this.yearDropdownSelector(this.id));

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/test.accessor.js:16:18)


 ✓ should handle month change with native dropdown element

 Datepicker - With Locale
 ✖ should display month in english

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         11 |    const dropDown = await Selector(this.selectMonthDropdownSelector(this.id));
         12 |    await this.t.click(dropDown);
         13 |    const selectedMonth = await Selector(
         14 |      this.selectAnyMonthSelector(this.id, numericMonth, native)
         15 |    );
       > 16 |    await this.t.click(selectedMonth);
         17 |  }
         18 |
         19 |  async chooseAnyYear(numericYear, native) {
         20 |    await this.t.setTestSpeed(0.5);
         21 |    const dropDown = await Selector(this.yearDropdownSelector(this.id));

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/test.accessor.js:16:18)


 Datepicker - Collapsible Version
 ✖ should write date into input field for input calendars

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         11 |    const dropDown = await Selector(this.selectMonthDropdownSelector(this.id));
         12 |    await this.t.click(dropDown);
         13 |    const selectedMonth = await Selector(
         14 |      this.selectAnyMonthSelector(this.id, numericMonth, native)
         15 |    );
       > 16 |    await this.t.click(selectedMonth);
         17 |  }
         18 |
         19 |  async chooseAnyYear(numericYear, native) {
         20 |    await this.t.setTestSpeed(0.5);
         21 |    const dropDown = await Selector(this.yearDropdownSelector(this.id));

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/test.accessor.js:16:18)


 ✖ should change enhanced dropdown title (only on large screens) on month change

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         11 |    const dropDown = await Selector(this.selectMonthDropdownSelector(this.id));
         12 |    await this.t.click(dropDown);
         13 |    const selectedMonth = await Selector(
         14 |      this.selectAnyMonthSelector(this.id, numericMonth, native)
         15 |    );
       > 16 |    await this.t.click(selectedMonth);
         17 |  }
         18 |
         19 |  async chooseAnyYear(numericYear, native) {
         20 |    await this.t.setTestSpeed(0.5);
         21 |    const dropDown = await Selector(this.yearDropdownSelector(this.id));

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/test.accessor.js:16:18)


 Datepicker React
 ✓ should render datepicker as reactified component
 ✓ should correctly expand year ranges assigned via property

 Datepicker React controlled
 ✓ datepicker should behave correctly when controlled

 Datepicker React inputfield
 ✓ should fire onDateChange callback on valid user input

 Datepicker React inputfield
 ✓ should react to programmatic date property changes

 Datepicker React empty inputfield
 ✖ should allow month change from default date

   1) The element that matches the specified selector is not visible.

      Browser: Chrome 64.0.3282.119 / macOS 10.13.6 (https://automate.browserstack.com/builds/8db88cf5ec1584d2c91ec13b825b8488460eece7/sessions/f333aa25dc01103b1d721420123392660f9c5ae8)

         399 |  const newMonth = currentMonth + (currentMonth === JANUARY ? 1 : -1);
         400 |
         401 |  // click on new month
         402 |  await t
         403 |    .wait(50)
       > 404 |    .click(
         405 |      `#datepicker-react .js-dropdown__button[data-index="${newMonth + 1}"]`
         406 |    );
         407 |
         408 |  // commit new date
         409 |  await t.wait(50).click(`#datepicker-react .js-datepicker__button-ok`);

         at <anonymous> (/Users/c304670/IdeaProjects/patterns-library/src/components/20-molecules/datepicker/ui.test.js:404:6)


 Datepicker React empty inputfield
 ✓ should pass through className and arbitrary data- attributes + have automatic React displayName
 ✓ should display error message if invalid is set, unless invaliddatetext is empty

 Datepicker Form
 ✓ should submit datepicker correctly in form
 ✓ should have default width

 Datepicker with onchange handler
 ✓ should fire the right events

 Datepicker as inputfield with fixed width and height
 ✓ should have 200px width
 ✓ button should have correct height
 ✓ input wrap should have correct height without a label

 Datepicker as inputfield with label
 ✓ component and input wrap should have correct height

 Datepicker as inputfield with 196px width and 10px height
 ✓ should have a minimum width
 ✓ should have a minimum height


 13/54 failed (7m 53s)

```

**Comment by danyball** [comment]
With `Safari 13.0.4 / iOS 13.3` all the datepicker tests are failing

**Comment by danyball** [comment]
Closed because we migrate to another test framework.

**Pull request closed without merging**